scripts/o3de/o3de/engine_template.py

```diff
@@ -2651,6 +2716,37 @@ def add_args(subparsers) -> None:
     create_gem_subparser.add_argument('--no-register', action='store_true', default=False,
                                       help='If the gem template is instantiated successfully, it will not register the'
                                            ' gem with the global, project or engine manifest file.')
+    create_gem_subparser.add_argument('-dn', '--display-name', type=str, required=False,
+                       help='The name displayed on the Gem Catalog')
+    create_gem_subparser.add_argument('-s', '--summary', type=str, required=False,
+                       default='A short description of this Gem',
+                       help='A short description of this Gem')
+    create_gem_subparser.add_argument('-req', '--requirements', type=str, required=False,
+                       default='Notice of any requirements for this Gem i.e. This requires X other gem',
+                       help='Notice of any requirements for this Gem i.e. This requires X other gem')
+    create_gem_subparser.add_argument('-l', '--license', type=str, required=False,
+                       default='License used i.e. Apache-2.0 or MIT',
+                       help='License used i.e. Apache-2.0 or MIT')
+    create_gem_subparser.add_argument('-lu', '--license-url', type=str, required=False,
+                       default='Link to the license web site i.e. https://opensource.org/licenses/Apache-2.0',
+                       help='Link to the license web site i.e. https://opensource.org/licenses/Apache-2.0')
+    create_gem_subparser.add_argument('-o', '--origin', type=str, required=False, 
+                       default='The name of the originator or creator',
+                       help='The name of the originator or creator i.e. XYZ Inc.')
+    create_gem_subparser.add_argument('-ou', '--origin-url', type=str, required=False, 
+                       default='The website for this Gem',
+                       help='The website for your Gem. i.e. http://www.mydomain.com')
+    create_gem_subparser.add_argument('-ut', '--user-tags', type=str, required=False,
+                       default='',
+                       help='A comma separated list of keywords.')
```

Comment:
The way the `gem_properties.py` script adds tags is a lot [more robust](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/gem_properties.py#L158-L159)

The python argparse configuration uses `nargs='*'`  to support multiple instances of --user-tags option.

Furthemore it splits the arguments on commas if it the input is a string otherwise it uses the [list as is](https://github.com/o3de/o3de/blob/development/scripts/o3de/o3de/gem_properties.py#L34-L36).